contracts/application/management/AppManager.sol

```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.4.21;
+
+import "./BaseAppManager.sol";
+import "../versioning/Package.sol";
+import "../../upgradeability/UpgradeabilityProxyFactory.sol";
+
+contract AppManager is BaseAppManager {
+  Package public package;
+  string public version;
+
+  function
+    AppManager(Package _package, string _version, UpgradeabilityProxyFactory _factory)
+    BaseAppManager(_factory)
+  public {
+    require(_package != address(0));
+    require(_package.hasVersion(_version));
+    package = _package;
+    version = _version;
+  }
+
+  function setVersion(string newVersion) public onlyOwner {
+    require(package.hasVersion(newVersion));
+    version = newVersion;
+  }
+
+  function getProvider() internal view returns (ContractProvider) {
```

Comment:
The rest of the AppManager contracts at this "user-level" use public members with auto generated getters, so this getProvider() could be removed.

Reply:
No given the `BaseAppManager` defines an abstract `getProvider` function which is resolved here or in `UnversionedAppManager`